api/src/main/resources/api/service_openapi.yaml

```diff
@@ -20,8 +20,149 @@ paths:
         500:
           description: Service is broken
 
+  # TODO implement "real" export endpoints
+  '/entityFilter/sqlQuery':
```

Comment:
1. Maybe. It very much depends on what the frontend prefers - shared resources or dynamicly retrieved from endpoints.
2. Probably around this same path, something like `:count` for counts, or `:ids`, or an "export file to google bucket" etc.
3. Totally up to product requirements. Sometimes the query, sometimes the results, I think. I'm thinking of the write to a Terra notebook as a "dataset export"
4. I don't think there's a lot, just counting, demographic counts, getting the ids, and generating queries. Eventually, with stateful artifacts, we'll need CRUD endpoints.

I'm not really familiar with http path best practices to really say. What do you think we should be doing here?
We could say this is about /entities/ and `filters` is a thing you can do on entities, along with `search` (which I think might have a few verbs under it). I don't know if dataset endpoints are also `entities` or just related enough.

We could do something like `/entities/{entityName}/filters:generateSqlQuery` or even `/underlays/{underlayName}/entities/{entityName}/filters:generateSqlQuery` but I don't know if that's an improvement.

Reply:
I suspect that ultimately, the API could look something like this - assuming a world where Tanagra is hosting multiple datasets / dataset versions:

### Main query endpoints
POST /datasets/{dataset_id}/entities:count # raw count
POST /datasets/{dataset_id}/entities:aggregate # chart data
POST /datasets/{dataset_id}/entities:generateSqlQuery # this PR

### Meta-schema endpoints
GET /datasets/{dataset_id}/entities
GET /datasets/{dataset_id}/entities/{entity_id}
...

Possibly more CRUD for the schema, if we wanted to support admin APIs.

Some of this might depend on what nouns eventually emerge for the persistent storage, e.g. if we're calling these things "cohorts" or something else.

Reply:
Thanks, sounds good.
So far I'm going with `underlay` instead of `datasets` following a brainstomr suggestion from the design doc. I'm not attached to that, but I'd like to avoid using "dataset" which is a very overloaded term.